### PR TITLE
TD-4383 Warn centre admins that active admin account exists when inactivating delegate account

### PR DESCRIPTION
**Pull request by sherif-olaboye**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-4383

### Description
I added a new button that only show if the delegate has admin role
I added confirmation page showing a list of centre admin roles and radio buttons to choose whether to deactivate only the delegate account or both the delegate and admin accounts
 I Implement logic to process user selection by deactivating delegate account only or both delegate account and admin account 
I added validation to ensure account is selected, also put post summit validation to ensure the page did not return if the use click back button after deactivation 
### Screenshots
image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [x] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            return RedirectToAction("Index", "ViewDelegate", new { deactivateDelegateAccountViewModel.DelegateId });
+            
+        }
+        private List<string>? GetRoles(AdminAccount? adminAccount, UserEntity userEntity)
+        {
+            var roles = new List<string>();
+            if (adminAccount != null)
+            {
+                var adminentity = new AdminEntity(adminAccount, userEntity.UserAccount, null);
+                CultureInfo currentCulture = System.Threading.Thread.CurrentThread.CurrentCulture;
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs:90]: why is this needed?

**Review by rshrirohit — COMMENTED** [review]
Please see inline comment.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs (hunk 1 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs]: removed the line of code

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
+﻿
+
+namespace DigitalLearningSolutions.Web.Controllers.TrackingSystem.Delegates
+{
+    using System.Collections.Generic;
+    using System.Globalization;
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs:6]: please remove this, it is not required here

**Review by rshrirohit — CHANGES_REQUESTED** [review]
Please address the inline change requested. Also, for any account that is not the current user, clicking deactivate account results in 404 error -
image

**Comment by sherif-olaboye** [comment]
> Please address the inline change requested. Also, for any account that is not the current user, clicking deactivate account results in 404 error - image

image
this is resolved

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs (hunk 2 above)
Comment (sherif-olaboye) [reply DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/DeactivateDelegateController.cs:6]: Done
